### PR TITLE
Simplify tSafe in query node

### PR DESCRIPTION
**Pull request by bigsheeper**
Signed-off-by: bigsheeper <yihao.dai@zilliz.com>

issues: #12952

### CODE REVIEW

File: internal/querynode/data_sync_service.go [hunk 1]
```diff
@@ -272,20 +271,6 @@ func (dsService *dataSyncService) startPartitionFlowGraph(partitionID UniqueID,
 func (dsService *dataSyncService) removePartitionFlowGraph(partitionID UniqueID) {
```
Comment (xige-16) [inline internal/querynode/data_sync_service.go:271]: why release partition fg is still needed?

File: internal/querynode/task.go [hunk 2]
```diff
@@ -931,26 +929,22 @@ func (r *releasePartitionsTask) Execute(ctx context.Context) error {
...
-						return err
-					}
-					sCol.removeVChannel(channel)
-					hCol.removeVChannel(channel)
+				r.node.tSafeReplica.removeTSafe(channel)
+				// no tSafe or tSafe has been removed,
+				// we need to remove the corresponding tSafeWatcher in queryCollection,
+				// and remove the corresponding channel in collection
+				qc, err := r.node.queryService.getQueryCollection(r.req.CollectionID)
+				if err != nil {
```
Comment (xige-16) [inline internal/querynode/task.go:937]: print error?

File: internal/querynode/tsafe.go [hunk 3]
```diff
@@ -54,133 +48,25 @@ func (watcher *tSafeWatcher) close() {
...
-	id UniqueID // collectionID or partitionID
-}
-
 type tSafe struct {
-	ctx         context.Context
-	cancel      context.CancelFunc
 	channel     Channel
 	tSafeMu     sync.Mutex // guards all fields
 	tSafe       Timestamp
 	watcherList []*tSafeWatcher
```
Comment (xige-16) [inline internal/querynode/tsafe.go:55]: why one channel has multi watcher?

File: internal/querynode/task.go (hunk 2 above)
Comment (bigsheeper) [reply internal/querynode/task.go:937]: it would  be printed in task scheduler

File: internal/querynode/data_sync_service.go (hunk 1 above)
Comment (bigsheeper) [reply internal/querynode/data_sync_service.go:271]: this pr doesn't contains the modifies of flow graph

File: internal/querynode/tsafe.go (hunk 3 above)
Comment (bigsheeper) [reply internal/querynode/tsafe.go]: fixed

**Pull request merged**